core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -644,8 +668,12 @@ class LogTest {
     assertEquals(2, log.latestProducerStateEndOffset)
 
     log.truncateTo(1)
-    assertEquals(None, log.latestProducerSnapshotOffset)
+    assertEquals(Some(1), log.latestProducerSnapshotOffset)
```

Comment:
This seems to be just an artifact of the refactor? I guess there is no harm taking a snapshot after every truncation since they are rare.

Reply:
Right, we now take at least one snapshot after truncation.